bin/alpen-client/README.md

```diff
@@ -0,0 +1,628 @@
+    participant G as Gossip
+    participant DB as Storage
+
+    BB->>OCT: Get pending inbox messages
+    OCT-->>BB: Vec<MessageEntry>
+
+    BB->>BB: Prepare BlockAssemblyInputs
+    BB->>PB: build_exec_payload()
+
+    PB->>PB: Execute transactions
+    PB->>PB: Apply deposits
+    PB->>PB: Extract withdrawals
+    PB-->>BB: Payload + WithdrawalIntents
+
+    BB->>BB: build_block_package()
+    BB->>DB: Store ExecBlockRecord
+    BB->>EC: new_block(hash)
+
+    EC->>EC: Update chain state
+    EC->>G: Broadcast header
+```
+
+---
+
+### Deposit Processing
+
+Deposits flow from Bitcoin L1 through OL into the EE:
+
+```
+┌─────────┐    ┌─────────┐    ┌─────────┐    ┌─────────┐
+│   L1    │    │   OL    │    │   EE    │    │   EVM   │
+│(Bitcoin)│    │         │    │ Client  │    │  State  │
+└────┬────┘    └────┬────┘    └────┬────┘    └────┬────┘
+     │              │              │              │
+     │ deposit tx   │              │              │
+     ├─────────────►│              │              │
+     │              │              │              │
+     │   finalized  │              │              │
+     │◄─────────────┤              │              │
+     │              │              │              │
+     │              │ inbox msg    │              │
+     │              ├─────────────►│              │
+     │              │              │              │
+     │              │              │ mint to addr │
+     │              │              ├─────────────►│
+     │              │              │              │
+```
+
+**Key Points**:
+- Only messages from finalized OL blocks are processed (to avoid reorg issues)
+- Deposits are rate-limited per block (`max_deposits_per_block`)
+- Subject ID (32 bytes) is truncated to EVM address (20 bytes)
+- Bitcoin amounts (sats) are converted to wei
+
+---
+
+### Withdrawal Processing
+
+Withdrawals flow from EE through OL back to Bitcoin:
+
+```
+┌─────────┐    ┌─────────────┐    ┌─────────┐    ┌─────────┐
+│   EVM   │    │   Payload   │    │   OL    │    │   L1    │
+│  State  │    │   Builder   │    │         │    │(Bitcoin)│
+└────┬────┘    └──────┬──────┘    └────┬────┘    └────┬────┘
+     │                │                │              │
+     │ withdrawal     │                │              │
+     │ intent         │                │              │
+     ├───────────────►│                │              │
+     │                │                │              │
+     │                │ BlockOutputs   │              │
+     │                │ (SentMessage)  │              │
+     │                ├───────────────►│              │
+     │                │                │              │
+     │                │                │    bridge    │
+     │                │                ├─────────────►│
+     │                │                │              │
+```
+
+**Status**: Withdrawal extraction is implemented. Batch submission to OL with SNARK proofs is not yet implemented.
```

Comment:
I think we should remove the status, since we are working on batch assembly already.

```suggestion
```